test/test-network/conf/25-wireguard.netdev

```diff
@@ -0,0 +1,12 @@
+[NetDev]
+Name=wg99
+Kind=wireguard
+
+[WireGuard]
+PrivateKey=EEGlnEPYJV//kbvvIqxKkQwOiS+UENyPncC4bF46ong=
+ListenPort=51820
+
+[WireGuardPeer]
+PublicKey=RDf+LSpeEre7YEIKaxg+wbpsNV7du+ktR99uBEtIiCA=
+AllowedIPs=fd31:bf08:57cb::/48,192.168.26.0/24
+Endpoint=wireguard.example.com:51820
```

Comment:
You can also add a second peer with an IP address here.
In the test there could be a check that verify that both peers
where added except that the endpoint for this peer was not set due 
an invalid hostname.

Reply:
This will need https://github.com/systemd/systemd/pull/10618/files#diff-f29d1bfc98e548dc0eb497c3d17cbefaR703
to finish without complaints about memory leaks.

Reply:
@Mic92 thank you! I've just added `FwMark`, `PresharedKey` and `PersistentKeepalive` as you suggested. Regarding the second peer, well, it's the first time I've installed `wireguard` on a VM so generally I have no idea what it is and what I'm doing here :-). I think it would be great if you could update the test later because I'm pretty sure you know what can and should be tested. Plus, maybe, there are publicly available URLs you're aware of that could be used to test the basic wireguard functionality or whatever else you think would useful.